### DISCUSSION TITLE
Don't create hard links for files that are not important

### DISCUSSION THREAD

**Discussion by zkochan** [pnpm/pnpm#3218] [Ideas]
Currently, all the files of a package are hard linked into. However, there are some files that are not used by code, they can be skipped during hard linking:

* markdown files: README.md, CHANGELOG.md, HISTORY.md, CONTRIBUTING.md
* all files in the `test/`, `tests/`, `example/`, `examples/` folders
* CI config files: `.travis.yml`, `.appveyor.yml`
* editor configs: `.editorconfig`
* npm/yarn stuff: `.npmrc`, `.npmignore`, `npm-shrinkwrap.yaml`, `yarn.lock`
* git things:`.gitignore`, `.gitattributes`, `.github/`
* LICENSE, LICENCE

Edit:

This features looks similar to [`yarn autoclean`](https://classic.yarnpkg.com/en/docs/cli/autoclean)

**Comment by kopach** [comment]
Hi @zkochan, any updates on this? How about just having `pnpm autoclean` which would do the same as [yarn autoclean](https://github.com/yarnpkg/yarn/blob/7cafa512a777048ce0b666080a24e80aae3d66a9/src/cli/commands/autoclean.js)? 
Additionally, it could be cool to have some parameter to `pnpm install` which would prevent those files being linked in the first place. This could be default on CI env

P.S.
As a temporary solution, for anyone looking at this, there is https://github.com/tj/node-prune
